### PR TITLE
Add the option to use analytical weights in 2-step diff-in-diff estimator

### PR DESCRIPTION
**Pull request by marcandre259**
Refers to #557 

Check that results are close to R's did2s effect's standard error and coefficient.

### CODE REVIEW

**Comment by marcandre259** [comment]
pre-commit.ci autofix

File: pyfixest/did/did2s.py [hunk 1]
```diff
@@ -82,13 +87,17 @@ def __init__(
         self._first_u = np.array([])
         self._second_u = np.array([])
 
+        # column name with weights. None by default
+        self._weights = weights
```
Comment (s3alfisc) [inline pyfixest/did/did2s.py:91]: For consistency with the `Feols` class, could we call this `self._weights_name`?

Comment (s3alfisc) [reply pyfixest/did/did2s.py:91]: Actually, not really a must have 🤔

**Comment by s3alfisc** [comment]
Took a first look and this looks great! Will take a second look and then likely merge tomorrow morning. Is the test data set published under a license that allows us to re-publish it?

File: pyfixest/did/did2s.py (hunk 1 above)
Comment (marcandre259) [reply pyfixest/did/did2s.py:91]: Hey, definitely good to add consistency. I'm going to add it.

**Comment by marcandre259** [comment]
> Took a first look and this looks great! Thanks @marcandre259! Will take a second look and then likely merge tomorrow morning. Is the test data set published under a license that allows us to re-publish it?

I'm not aware of a license associated with the data. Here is some background info that may be useful in deciding whether it can be hosted for testing. 

It is data published by the American government (.gov) and cleaned by the authors below. It's referenced in Scott Cunningham's the mixtape and taken from Cheng Cheng and Mark Hoestra (2013). He says:
"The data they used is somewhat standard in crime studies. They used the FBI Uniform Crime Reports Summary Part I files from 2000 to 2010"

The data comes from:
DOES STRENGTHENING SELF-DEFENSE LAW DETER CRIME OR ESCALATE
VIOLENCE? EVIDENCE FROM CASTLE DOCTRINE
By Cheng Cheng and Mark Hoestra (2013)

Who point out the following:
Outcome data come from the FBI Uniform Crime Reports (UCR) and cover all 50
states from 2000 – 2010. 10 Specifically, we use homicide, burglary, robbery, and
aggravated assault data from the official UCR data published online by the FBI. 11 In
addition, for the other variables not available from the online UCR, we use data from the
FBI’s Master files (Return A and Supplemental Homicide Report).

**Comment by s3alfisc** [comment]
@marcandre259 I dropped the new test data set (even though it is likely ok to include it, for example it is distributed via the R `did2s` package) and refactored the did2s tests - they now all run via the `df_het` data set. Will merge after the CI passes. Thanks for implementing this!

**Review by s3alfisc — APPROVED** [review]

**Pull request merged**